generators/app/templates/src/app/screens/OnBoarding/components/Swiper/index.js

```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react';
+import { SafeAreaView } from 'react-native';
+import Swiper from 'react-native-swiper';
+import PropTypes from 'prop-types';
+
+import screens from './screens';
+import Footer from './components/Footer';
+import styles from './styles';
+
+class CustomStepSwipper extends PureComponent {
+  state = {
+    scrollIndex: 0
+  };
+
+  handleNextScreen = () => {
+    this.scrollView.scrollBy(1);
+    this.setState(prevState => ({ scrollIndex: prevState.scrollIndex + 1 }));
+  };
+
+  handlePreviousScreen = () => {
+    this.scrollView.scrollBy(-1);
+    this.setState(prevState => ({ scrollIndex: prevState.scrollIndex - 1 }));
+  };
+
+  handleChangeIndex = index => this.setState({ scrollIndex: index });
+
+  handleRef = ref => (this.scrollView = ref);
```

Comment:
Why aren't you using 'Hooks' here? I think this is wrong

Reply:
@ignaciosantise @guidoprinc What do you think? I think we shouldn't use hooks always, it depends on the case, I don't like the idea of stop using Component classes, I think we should mix Hooks and Components.

Reply:
@felire ok, why do you think we should use classes sometimes? got strong reasons?

Reply:
Maybe this is not the case because is a simple component. But in some cases I think we should avoid have functions with a lot of hooks, a lot of `useEffect`, `useCallback` and `useState`. I think that if a function with many hooks grow a lot it's more difficult to read. And I'm not sure about the performance, is using hooks really more performant than classes? Do not misunderstand me, I'm not against of hooks, in fact I like using them, but I don't like the idea of avoid classes in all cases.

Reply:
I moved this to hooks finally because the component is simple, but my doubt is more general